### ISSUE TITLE
[HOLD for payment 2024-12-05] [$250] Onbording - Onboarding input is grayed out when select auto suggested name

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51523]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.54
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** gocemate+a2663@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Login with new user
3. When onboarding modal opens select Something else
4. Start typing until auto suggested name appears
5. Select the suggested name

## Expected Result:
Selected name should be black

## Actual Result:
Name input is grayed out when select auto suggested name

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/da5b2c72-2496-4202-983b-1fdba7141c32

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021851668566536953494</li>
        <li>Upwork Job ID: 1851668566536953494</li>
        <li>Last Price Increase: 2024-11-06</li>
<li>Automatic offers: </li>
<ul>
<li>klajdipaja | Contributor | 104956195</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @JmillsExpensify</details>

**Comment by JmillsExpensify** [comment]
Opening up to the community.

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Selected auto-suggested name or last name appears grayed out in the input field, returning to black only after the user modifies the text.
### What is the root cause of that problem?
The root cause is that browser's default auto-fill/auto-suggest styling is overriding our custom text color styles
### What changes do you think we should make in order to solve the problem?
We should add the `WebkitTextFillColor:theme.text` to our `baseTextInput` styles 
Here:
https://github.com/Expensify/App/blob/aa6401de4a3c02adce2cb898eba8c58e1c92b62d/src/styles/index.ts#L1226-L1235

**Comment by Shahidullah-Muffakir** [comment]
https://github.com/user-attachments/assets/cda8ad4a-2e0b-4559-be57-f75e9f1dac6c

**Comment by s77rt** [comment]
Not able to reproduce


https://github.com/user-attachments/assets/871f0631-8ff3-4f47-a116-af78168d44d9

**Comment by s77rt** [comment]
@Shahidullah-Muffakir Were you able to reproduce the bug? Any additional required steps?

**Comment by klajdipaja** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Input fields on the onboarding page (name and last name) have grayish color after the user loges in to expensify.
To reproduce this bug I had to set the theme in my MacOS to Light in addition to the steps by the reporter.

### What is the root cause of that problem?
The root cause is that the global autofill-input styles that we apply in the `ThemeProvier.tsx` in this block:
https://github.com/Expensify/App/blob/664e657f3c5d5f1ce4c55d09eee054519228432e/src/components/ThemeProvider.tsx#L24-L26

The login page initiates a `ThemeProvider` with a static override in Dark mode and after the user is logged in the main ThemeProvider in `App.tsx` should be based on the system mode as Light. 

The `ThemeProvider` does change to Light mode as we can see from all the styles that change in the main page. The problem is that since we are adding the styles in the `ThemeProvider` itself the styles that are added globally are of the latest mounted `ThemeProvider` and turnes out the last one is with a static override theme.
 
### What changes do you think we should make in order to solve the problem?

The `useEffect` is not consistent with the way we use themes and should not be in the `ThemeProvider` but in one of the inner wrappers in `App.tsx`.
To me it makes most sense to move the `useEffect` in `ColorSchemeWrapper` which solves the issue we have here.



### What alternative solutions did you explore? (Optional)

**Comment by Shahidullah-Muffakir** [comment]
@S77rt Yes, I was able to reproduce the bug in both Safari and Chrome.
Note: The issue does not occur if we refresh the page.

Safar: 

https://github.com/user-attachments/assets/40c1c370-454d-4738-b4b9-d99427c4df1f

macOS Chrome: 

https://github.com/user-attachments/assets/9439a44f-a628-4be6-b4e1-42a582d001b5

**Comment by s77rt** [comment]
@Shahidullah-Muffakir Thanks for the confirmation. Regarding the proposal, your RCA does not look correct as we do apply our custom auto-fill styles and seeing that it works on refresh confirms it.

**Comment by s77rt** [comment]
@klajdipaja Thanks for the proposal. Can you please elaborate on the RCA? Is it the double call to `DomUtils.addCSS` that causes the bug? It's not clear why the bug occurs given that the styles are overwritten

**Comment by klajdipaja** [comment]
@s77rt the double call is not the issue. It's the fact the we are loading these styles in the `ThemeProvider` and that opens the app up for race conditions when there are multiple `ThemeProviders` in the app. I added some `console.log` in the `ThemeProvider` and noticed that the last render of the provider has a `staticThemePreference` passed to it as DARK which is only done in the `SignInPage.tsx`: `<ThemeProvider theme={CONST.THEME.DARK}>` making it so that the latest styles added are actually for DARK mode.

**Comment by s77rt** [comment]
@klajdipaja Is it that the `useEffect` that runs last in `ThemeProvider` has the wrong theme? (`staticThemePreference`) If so, why this does not happen when we login with an existing account?

<img width="633" alt="Screenshot 2024-10-31 at 6 43 13 PM" src="https://github.com/user-attachments/assets/760db5d5-e6d1-47fb-9ff9-e2419f556515">

**Comment by klajdipaja** [comment]
> Is it that the useEffect that runs last in ThemeProvider has the wrong theme? (staticThemePreference)
Yes, that's right. Which is then loading the global style with text color gray which is for dark mode.
This in my case happens for both login types (new and existing user) which I see by checking the style that is loaded in inspect element. Search for it in inspect element  by  `id="autofill-input"`. In both cases for me I see the text color set to `#E7ECE9`.

Check this screenshots:
![Screenshot 2024-11-01 at 11 30 34 AM](https://github.com/user-attachments/assets/008fd0f0-1c75-43b6-8787-799fbe739add)
![Screenshot 2024-11-01 at 11 32 00 AM](https://github.com/user-attachments/assets/9a460b7d-9717-4dd3-b17e-c823bfcd7601)

**Comment by s77rt** [comment]
@klajdipaja I'm still having trouble reproducing this. Is there some additional steps that I'm missing?

<img width="1728" alt="Screenshot 2024-11-01 at 2 46 58 PM" src="https://github.com/user-attachments/assets/81b76869-ed84-445e-90c6-c1d033a94603">

**Comment by klajdipaja** [comment]
@s77rt I can reproduce this all the time if I have the System->Appearance settings of my macOS set to Light.

**Comment by s77rt** [comment]
@klajdipaja Ah I see. Now i can reproduce. Isn't the `ThemeProvider` supposed to re-render when the theme changed?

**Comment by klajdipaja** [comment]
@s77rt It should re-render. That's what I am experiencing in a test I just did now

**Comment by s77rt** [comment]
@klajdipaja I don't think we have reached a clear root cause yet. The `ThemeProvider` from my end didn't re-render when the theme changed from forced dark to light (system). If it did re-render on your side, why the `useEffect` didn't execute?

**Comment by klajdipaja** [comment]
Hi @s77rt I have looked more into it and read about the Context API too to fully understand how it works and I think I have come up with a definitive RCA.
We have two `ThemeProviders` in the Expensify app:
1. The main one: in App.tsx https://github.com/Expensify/App/blob/0647baf7b70967a707df6eae6cb4c3f5a260a8f5/src/App.tsx#L73
2. Local one for Sign In page with static theme in `SingInPage.tsx`:
https://github.com/Expensify/App/blob/0647baf7b70967a707df6eae6cb4c3f5a260a8f5/src/pages/signin/SignInPage.tsx#L348

 `Context Providers` render only when there is a change in their state which for the theme provider it is depending on the state of `useThemePreferenceWithStaticOverride` hook which itself depends on the state of the hook `useThemePreference`  which depends on two states:
 a. `preferredThemeFromStorage` which is the stored value in Onyx. default is system.
 b. state of `useColorScheme()` hook which is a react hook that returns the preferred theme of the system.
 So, basically the `ThemeProvider` would re render if there is a change in the preferred user theme on Expensify settings or if the system theme changes.
 
 What happens when the user opens Expensify if not previously authenticated?
 1. The Main ThemeProvider loads first and sets the theme for all children to `system` and loads the static autofill css for `system` theme. (text color black)
 2. The local SignIn ThemeProvider then renders with the constant theme `DARK` and adds the static autofill css with `DARK` theme. (text color gray) 
3. When the user types the email and inputs the OTP the local theme provider renders again as it is forced by the re-render of the SignIn component. Which makes the last to be added static CSS the ones from the local ThemeProvider.


This behaviour is visible to new users and existing ones. You can test for existing users by logging in and then going to the Settings and trying to change the Name of the user. You will see the same gray color text as in the on-board modal.

> The ThemeProvider from my end didn't re-render when the theme changed from forced dark to light (system). If it did re-render on your side, why the useEffect didn't execute?

In my case it does execute the useEffect.
The change would cause a re-render if the value of the hook `useThemePreference` changes which depends on primarly on the preferred theme of the user. If the user does not have "use device settings" in the preferences then the system theme would not cause a change because it is ignored. But I don't think we should not focus on that because  it is not related to the bug.

In the screen recording below you can see that no rendering happens for the main ThemeProvider until I change the system settings and you can see that for existing users we have the same bug.


https://github.com/user-attachments/assets/88c9b210-e00b-4718-9dc4-bd3237be03ec

**Comment by s77rt** [comment]
@klajdipaja After you sign in, isn't the main ThemeProvider supposed to re-render?

**Comment by klajdipaja** [comment]
@s77rt No, not necessarily. It would re-render if there was a difference in the state which mainly comes from the Onyx `preferredTheme` key.

**Comment by s77rt** [comment]
@klajdipaja Okay, so what caused the theme to actually change? i.e. what component did re-render after sign in so that the theme switched to light

**Comment by klajdipaja** [comment]
@s77rt All the children components that use the ThemeContext (through `useTheme` hook) where re-rendered as they received a new value from the ThemeContext. They are now wrapped only by the main ThemeProvider so the value changed from the DARK of the local ThemeProvider to the system which then depends on your system config but Light for our test scenarios.

**Comment by s77rt** [comment]
@klajdipaja But isn't the value that we get from the `useTheme` hook coming from the `ThemeProvider`? Meaning that before the components (consumers) re-render the provider must have re-rendered first.

**Comment by klajdipaja** [comment]
@s77rt The theme value comes from the closeset ThemeProvider in the tree. 

The main provider has rendered when the app is opened  and since the state does not change it does not re-render. The consumer get the update because now they're reading the theme value from a different  ThemeProvider

**Comment by s77rt** [comment]
@klajdipaja Okay I think I got it. It seems that the render logic is as follow:
- Main ThemeProvider (system)
- Main ThemeProvider (system) Adds Autofill CSS 
- SignIn ThemeProvider (dark)
- SignIn ThemeProvider (dark) Adds Autofill CSS
- SignIn ThemeProvider (dark) is removed

Now components will use the Main ThemeProvider (system) with light theme but the Autofill CSS is still from dark theme.

Does that sound correct?

**Comment by klajdipaja** [comment]
Yes that's correct. I would change my solution to move the useEffect to `Expensify.tsx` instead of the `ColorSchemeWrapper` as that is used in other places too

**Comment by klajdipaja** [comment]
@s77rt I am not in favor of that solution for these reasons:
1. As the existing solution does too, I think it brings the Single Responsibility Principle of the ThemeProvider. The purpose of the Context API is to provide a consistent value to all the child components. It's then the responsibility of those components to use the context value as they need to. The fact that the ThemeProvider is both generating the value and manipulating the DOM with it I think breaks the purpose of the Context API.
2. If I understood your idea would to have for each ThemeProvider an id that could either be a random string or a prop of the provider component itself and then we would have something like this:
```
    useEffect(() => {
        DomUtils.addCSS(DomUtils.getAutofilledInputStyle(theme, `.${cssClass}`), `${id}-autofill-input`);
        return () => {
            DomUtils.removeCSS(`${id}-autofill-input`);
        };
    }, []);
```
Which would mean that in the Sign Page we would have two `-autofill-input` styles with different colors. The browser than would decide which style to apply and in this case that would depend on the order in which this both styles appear in the DOM. I don't think we can guarantee the order of that style, can we?

**Comment by s77rt** [comment]
@klajdipaja Can we move the `useEffect` to `ThemeStylesProvider`?

**Comment by klajdipaja** [comment]
@s77rt That's much better.
This is what I tried and works great:

```
    useEffect(() => {
        const randomId = Math.random().toString(36).slice(2, 7);
        const styleid = `${randomId}-autofill-input`;

        DomUtils.addCSS(DomUtils.getAutofilledInputStyle(theme.text), styleid);
        return () => {
            DomUtils.removeCSS(styleid);
        };
    }, [theme.text]);

```

**Comment by s77rt** [comment]
@klajdipaja I meant to just move the `useEffect` without adding the id as `ThemeStylesProvider` uses the `useTheme` hook and will re-render if the theme did change

**Comment by klajdipaja** [comment]
@s77rt That also works. Without a cleanup function, though.
```
    useEffect(() => {
        DomUtils.addCSS(DomUtils.getAutofilledInputStyle(theme.text), "autofill-input");
    }, [theme.text]);
```

**Comment by s77rt** [comment]
@klajdipaja Great! Can you please update your proposal to reflect the new solution?

**Comment by klajdipaja** [comment]
Thanks for the reminder Melvin. 
Proposal updated https://github.com/Expensify/App/issues/51523#issuecomment-2449570972  yesterday with the latest solution but forgot to add the comment

**Comment by s77rt** [comment]
@klajdipaja Thanks for the update! The proposal looks good to me.

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/51523#issuecomment-2449570972)

**Comment by s77rt** [comment]
@bondydaa Can you please review the approved proposal https://github.com/Expensify/App/issues/51523#issuecomment-2460228854?

**Comment by s77rt** [comment]
🎀 👀 🎀 To get another engineer

**Comment by s77rt** [comment]
@francoisl We have this approved proposal https://github.com/Expensify/App/issues/51523#issuecomment-2460228854. Please review and assign the contributor if looks good to you

**Comment by klajdipaja** [comment]
@s77rt I tested again with the proposed solution that we decided on and unfortunately it is not working as expected for me.
So I tried the other approach that we discussed here earlier and that one worked. 
Here's a recording of two scenarios, first is with a new account and the other an old account with dark mode as a preference.

https://github.com/user-attachments/assets/191fe6eb-a1e0-43f5-b23e-72d732ce19d8

**Comment by s77rt** [comment]
@klajdipaja Can you please explain why it's not working as expected? The `ThemeStylesProvider` component should re-render since it's using the `useTheme` hook and the effect should overwrite the autofill styles. Am I missing something here?

**Comment by klajdipaja** [comment]
@s77rt It does not re-render post login because the value provided by `useTheme` did not change. The main ThemeProvider had the value theme=light because the preference in onyx was `system`. The onyx value would have to change for the useTheme to change it's value again. For an already registered user with `dark` as a preference I do get the re-render.

The global css seem very buggy and unpredictable, we have had another issue with the Sign In input where we added another workaround (reference issue-> https://github.com/Expensify/App/issues/50460).
I actually found a very clean solution that we have used already used in the style defintion for other elements but maybe it was missed when the addCss approach was added. 

We have a consistent use of the theme vavlues by adding the styles here:
https://github.com/Expensify/App/blob/7bfb34527508b47732cab6d4280a943ffab48066/src/styles/index.ts#L1226

Here is what  I tried in my draft PR that seems to work on all platforms:

https://github.com/klajdipaja/Expensify-App/blob/fb5bdfa37d89aae7c8a14f9dd21967aa320eaafe/src/styles/index.ts#L1226-L1243

Here's a screen recording with the change including removing the use of addCSS from the SignIn page:

https://github.com/user-attachments/assets/bfd2fa47-394b-4832-bdc4-e57d68dfccee

**Comment by s77rt** [comment]
@klajdipaja Since `getAutofilledInputStyle` is called from `SignInPageLayout` we should avoid calling it from `ThemeProvider`. Will that fix our issue?

**Comment by klajdipaja** [comment]
@s77rt I removed the use of `getAutofilledInputStyle` from both the `SignInPageLayout` and `ThemeProvide` and added them `into styles/index.ts`:
```
/ These properties are available in browser only
            ...(Browser.getBrowser()
                ? {
                      WebkitBackgroundClip: 'text',
                      WebkitTextFillColor: theme.text,
                      caretColor: theme.text,
                  }
                : {}),
        },
```
This to me seems like the best way to fix the bug because it uses the values provided by the theme directly in the style itself and we have not had any issues with the theme not providing correct values to the styles.

With those changes I am getting consistent results that solve the issue we have here and the one with the SIgnIn page. You can check those changes out in this Draft PR  https://github.com/Expensify/App/pull/52785.

In the first commit of the PR I had tried another solution that we had discussed here previously by adding the styles on mount and removing them on the unmount of the component. This is the commit ->
https://github.com/Expensify/App/pull/52785/commits/bcf3b41dc07073180a3b9d0ab63ba8124eec072b

I don't like that way as it seems like another workaround.

**Comment by s77rt** [comment]
This is affecting the placeholder too

| Before | After |
|:------:|:-----:|
|   <img width="523" alt="Screenshot 2024-11-20 at 2 55 53 PM" src="https://github.com/user-attachments/assets/ca2faf52-6dca-4044-b7d3-eaf7444221aa">     |     <img width="523" alt="Screenshot 2024-11-20 at 2 55 45 PM" src="https://github.com/user-attachments/assets/02121697-f804-4fab-af4c-48aba5c42045">  |

**Comment by klajdipaja** [comment]
@s77rt ahh you are right it does that because it applies the style even when the placeholder or input does not come from the autofill.
I worked on an improvement to that to add the style dynamically and only when the input is autofilled but since we cannot use pseudoclass selectors or attributes as there are used in the `getAutofilledInputStyle` I built a workaround.

I re-added the `getAutofilledInputStyle` in the `ThemeStylesProvider` but added in its styles this section:
```
    input[chrome-autofilled],
    input[chrome-autofilled]:hover,
    input[chrome-autofilled]:focus,
    input:-webkit-autofill,
    input:-webkit-autofill:hover,
    input:-webkit-autofill:focus,
    textarea[chrome-autofilled],
    textarea[chrome-autofilled]:hover,
    textarea[chrome-autofilled]:focus,
    textarea:-webkit-autofill,
    textarea:-webkit-autofill:hover,
    textarea:-webkit-autofill:focus{
        -webkit-background-clip: text;
        animation-name: onAutoFillStart;
        animation-duration: 0.01s;
    }
    @keyframes onAutoFillStart {
        from {}
        to {}
    }
```
What I do here is to trigger a no-op animation `onAutoFillStart` when ever one of the autofill selectors are active.
I then use that in `BaseTextInput/index.tsx` to set a state that the input or placeholder is from the autofill of the browser:
```
    const [isAutoFilled, setIsAutoFilled] = useState(false);
```

Set the `isAutoFilled` when the input change as the user types in new input or selects one option from the autofill dropdown

```
    const onChange = (event: NativeSyntheticEvent<TextInputFocusEventData>) => {
        inputProps.onChange?.(event);
        setIsAutoFilled(isInputAutoFilled(event.target));
    };
```

And use the  `onAutoFillStart` animation that we added in the CSS to detect if we have to do with an autofill interaction from the user and set the `isAutoFilled` state appropriately.
```
    // listen for the custom onAutoFillStart event to detect when the input is autofilled by the browser
    useEffect(() => {
        const handleAnimationStart = (event: AnimationEvent) => {
            if (event.animationName === 'onAutoFillStart') {
                setIsAutoFilled(true);
            }
        };
        const debouncedHandleAnimationStart = debounce(handleAnimationStart, 100);

        const inputElement = input.current;
        inputElement?.addEventListener('animationstart', debouncedHandleAnimationStart);

        return () => {
            inputElement?.removeEventListener('animationstart', debouncedHandleAnimationStart);
        };
    }, []);

```

 Now we apply the conditional style to the styles of the `InputComponent` here:
 https://github.com/Expensify/App/blob/4c44827398f884af626872c33f64dd358d19eff4/src/components/TextInput/BaseTextInput/index.tsx#L376
 By adding:
 ```
      isAutoFilled && styles.baseTextInputAutofill,
 ```
 And `baseTextInputAutofill` would be defined by using theme values in `styles/index.tsx`:
```  
baseTextInputAutofill:  {
            WebkitBackgroundClip: 'text',
            WebkitTextFillColor: theme.text,
            caretColor: theme.text,
        },
```

All this changes are available in the draft PR -> https://github.com/Expensify/App/pull/52785

**Comment by s77rt** [comment]
@klajdipaja That solution is over engineered and it's not fixing the root cause. We better keep the global css and focus on the useEffect calls.

**Comment by klajdipaja** [comment]
@s77rt I don't necessarily see it as over-engineering, more like compensating lack of flexibility in the styling capabilities of react Stylesheet.  The root cause is unavoidable if we add a global css as there is no guarantee to which ThemeProvider will render last and "win" with the styles it provides over other ThemeProvider. In the case of the SignIn issue, the Main ThemeProvider was rendering last due to the async operations that cause the re-render of the component.

But I'm fine with using the useEffects, like in this commit https://github.com/Expensify/App/pull/52785/commits/bcf3b41dc07073180a3b9d0ab63ba8124eec072b

**Comment by s77rt** [comment]
@klajdipaja Can we just not call the `getAutofilledInputStyle` from `ThemeProvider` if `staticThemePreference` is passed?

**Comment by klajdipaja** [comment]
@s77rt yes we can definitely do that, I just tested it and it works fine. 
The theme provider then imposes other components that us it to define those global styles in a way that works for them, like we did for the SignInPage. Kinda strange imposition but it works

**Comment by s77rt** [comment]
@klajdipaja Let's work on that and see how it goes

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by klajdipaja** [comment]
Will prepare the PR on Monday

**Comment by s77rt** [comment]
# BugZero Checklist:

- [X] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [X] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [X] 2a. Reported on production
  - [ ] 2b. Reported on staging (deploy blocker)
  - [ ] 2c. Reported on both staging and production
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [X] 3d. QA
  - [ ] 3z. Other:

</details>

- [X] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: https://github.com/Expensify/App/pull/35242#discussion_r1864971101

- [X] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: n/a

- [X] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

    Bug requires regression test: Yes

<summary>Regression Test Proposal Template</summary>

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


- System theme is light

### Test:


1. Login with a new user
2. After the onboarding modal is displayed, select "Something else"
3. Start typing in the first name field until auto suggested names appear
4. Select any suggested name
5. Verify that the suggested name font color is dark and the name is readable

Do we agree 👍 or 👎

**Comment by JmillsExpensify** [comment]
Payment summary:

 - Contributor: $250 @klajdipaja via Upwork
 - Reviewer: $250 @s77rt via New Expensify

**Comment by JmillsExpensify** [comment]
Contributor paid via Upwork, regression test created, and remaining payments are via New Expensify. Closing this one!

**Comment by garrettmknight** [comment]
**Payment summary:**

- Contributor: $250 @klajdipaja via Upwork
- Reviewer: $250 @s77rt via New Expensify

**Comment by JmillsExpensify** [comment]
$250 approved for @s77rt
